test/lib/jdk/test/lib/hprof/parser/Reader.java

```diff
@@ -166,7 +165,6 @@ public static String getStack(String heapFile, int debugLevel)
                 return r.printStackTraces();
             } else if ((i >>> 8) == GZIP_HEADER_MAGIC) {
                 // Possible gziped file, try decompress it and get the stack trace.
-                in.close();
```

Comment:
It is not obvious to me that there may not be a reason for closing all of the streams before opening new ones below.